packages/typespec-ts/src/utils/operationUtil.ts

```diff
@@ -437,3 +449,133 @@ export function getCollectionFormatHelper(
   const detail = getSpecialSerializeInfo(paramType, paramFormat);
   return detail.descriptions.length > 0 ? detail.descriptions[0] : undefined;
 }
+
+export async function appendDefaultResponseIfAbsent(
+  dpgContext: SdkContext,
+  client: SdkClient
+) {
+  const operationGroups = listOperationGroups(dpgContext, client);
+  const program = dpgContext.program;
+  for (const operationGroup of operationGroups) {
+    const operations = listOperationsInOperationGroup(
+      dpgContext,
+      operationGroup
+    );
+
+    for (const op of operations) {
+      const route = ignoreDiagnostics(getHttpOperationWithCache(program, op));
+      // ignore overload base operation
+      if (route.overloads && route.overloads?.length > 0) {
+        continue;
+      }
+      await appendIfAbsent(route);
+    }
+  }
+  const clientOperations = listOperationsInOperationGroup(dpgContext, client);
+  for (const clientOp of clientOperations) {
+    const route = ignoreDiagnostics(
+      getHttpOperationWithCache(program, clientOp)
+    );
+    if (route.overloads && route.overloads?.length > 0) {
+      continue;
+    }
+    await appendIfAbsent(route);
+  }
+}
+
+async function appendIfAbsent(route: HttpOperation) {
+  const errors = route.responses.filter((r) =>
+    isDefaultStatusCode(r.statusCode)
+  );
+  if (errors.length === 0) {
+    route.responses.push(FAKE_CORE_ERROR_RESPONSE as any);
+  }
+}
+
+export async function createRLCEmitterTestHost() {
+  return createTestHost({
+    libraries: [
+      HttpTestLibrary,
+      RestTestLibrary,
+      VersioningTestLibrary,
+      AzureCoreTestLibrary,
+      SdkTestLibrary
+    ]
+  });
+}
+const FAKE_CORE_ERROR_RESPONSE_TYPE = {
+  kind: "Model",
+  name: "ErrorResponse",
+  namespace: {
+    kind: "Namespace",
+    name: "Foundations",
+    namespace: {
+      kind: "Namespace",
+      name: "Core",
+      namespace: {
+        kind: "Namespace",
+        name: "Azure"
+      }
+    }
+  },
+  properties: [],
+  sourceModel: {
+    kind: "Model",
+    name: "ErrorResponseBase"
+  }
+};
+const FAKE_CORE_ERROR_RESPONSE = {
+  statusCode: "*",
+  type: FAKE_CORE_ERROR_RESPONSE_TYPE,
+  responses: [
+    {
+      body: {
+        contentTypes: ["application/json"],
+        type: FAKE_CORE_ERROR_RESPONSE_TYPE
+      }
+    }
+  ]
+};
+
+const _cache = new Map();
+export function getHttpOperationWithCache(
+  program: Program,
+  operation: Operation
+): [HttpOperation, readonly Diagnostic[]] {
+  const existing = _cache.get(operation);
+  if (existing) {
+    return existing;
+  }
+  const httpOperationRef = getHttpOperation(program, operation);
```

Comment:
Every time `getHttpOperation` would build a new http operation according to original one so we need to cachec them otherwise the appending info would be lost.